python/cuml/internals/api_context_managers.py

```diff
@@ -66,10 +66,31 @@ def _using_mirror_output_type():
 
 
 def in_internal_api():
+    """
+    Checks if the code is currently executing within an internal API context.
```

Comment:
For this kind of call, it would be good to provide some context on exactly what this means. The name of the method is already fairly descriptive, so it is more important to emphasize _why_ we care about this. For this particular case, it is important because once we are making calls internal to a cuML API call, we have already performed the necessary conversion to a CumlArray, and we want to avoid extra copies to device.